lib/hnvm/config.sh

```diff
@@ -12,7 +12,42 @@ while [ -h "$source" ]; do # resolve $source until the file is no longer a symli
 done
 script_dir="$( cd -P "$( dirname "$source" )" >/dev/null 2>&1 && pwd )"
 
-export COMMAND_OUTPUT=${HNVM_OUTPUT_DESTINATION:-/dev/stdout}
+COMMAND_OUTPUT=""
+
+# Try aggressively to find an environment-supported, redirectable, stdout fd descriptor
+#
+# In nodejs spawned subprocess (think npm pre/postinstall scripts), stdio can be
+# provided as sockets, which is not able to do simple '>>' bash redirection
+# https://nodejs.org/api/child_process.html#optionsstdio
+# "These are not actual Unix pipes and therefore the child process can not use them by their
+# descriptor files, e.g. /dev/fd/2 or /dev/stdout."
+#
+# For command output shell redirection, hierarchy is (most to least priority)
+# - $HNVM_OUTPUT_DESTINATION - arbitrary path given as env var
+# - /dev/stdout - convenience symlink to fd 1 - not fully posix portable
+# - /dev/fd/1 - on macos, the character special device (c in `ls`) that is the stdout stream
+# - /dev/null - only used as fallback, if none of the above exist
```

Comment:
I think all of the comments here are actually making the code harder to read. I think maybe we pull them out and just add this detail to the PR as comments. You can always use git blame to see the discussion and reasoning. At most I'd be comfortable with Pointing to a ticket, PR, or stackoverflow file or something explaining the issue

Reply:
Sounds good, I will rework this

Reply:
done